src/coordinator/mod.rs

```diff
@@ -14,11 +15,25 @@ pub type InputsStore = std::collections::HashMap<Serial, lxp::packet::ReadInputs
 pub struct Coordinator {
     config: ConfigWrapper,
     channels: Channels,
+
+    inverter_receiver: std::cell::Cell<Option<inverter::Receiver>>,
+    mqtt_receiver: std::cell::Cell<Option<mqtt::Receiver>>,
 }
 
 impl Coordinator {
     pub fn new(config: ConfigWrapper, channels: Channels) -> Self {
-        Self { config, channels }
+        // Create the receivers at new() time, rather than start(), so things
+        // wanting to send to us will queue messages instead of failing with an
+        // error about closed channels
+        let inverter_receiver = std::cell::Cell::new(Some(channels.from_inverter.subscribe()));
```

Comment:
The coordinator's commands also have receivers on `from_inverter`, which are created as needed in the `run` methods rather than at initialization. I've left those alone, since they don't seem to suffer from the same kind of problem.